src/auth-service/controllers/candidate.js

```diff
@@ -1,45 +1,52 @@
-const Candidate = require("../models/Candidate");
+const CandidateSchema = require("../models/Candidate");
 const HTTPStatus = require("http-status");
 const msgs = require("../utils/email.msgs");
 const validateCandidateInput = require("../utils/validations.candidate");
 const register = require("../utils/register");
+const { getModelByTenant } = require("../utils/multitenancy");
+
+const CandidateModel = (tenant) => {
+  return getModelByTenant(tenant, "candidate", CandidateSchema);
+};
 
 const candidate = {
-    registerCandidate: (req, res) => {
-        console.log("yay!");
-        console.log("the elements we need:");
-        console.dir(req.body);
-        const { errors, isValid } = validateCandidateInput(req.body);
-        if (!isValid) {
-            return res
-                .status(400)
-                .json({ success: false, errors, message: "validation error" });
-        }
+  registerCandidate: (req, res) => {
+    const { tenant } = req.query;
+    console.log("yay!");
+    console.log("the elements we need:");
+    console.dir(req.body);
+    const { errors, isValid } = validateCandidateInput(req.body);
+    if (!isValid) {
+      return res
+        .status(400)
+        .json({ success: false, errors, message: "validation error" });
+    }
 
-        const mailOptions = {
-            from: `info@airqo.net`,
-            to: `${req.body.email}`,
-            subject: "AirQo Platform JOIN request",
-            text: msgs.joinRequest,
-        };
+    const mailOptions = {
+      from: `info@airqo.net`,
+      to: `${req.body.email}`,
+      subject: "AirQo Platform JOIN request",
+      text: msgs.joinRequest,
+    };
 
-        register(req, res, mailOptions, req.body, Candidate);
-    },
+    register(req, res, mailOptions, req.body, CandidateModel(tenant));
+  },
 
-    getAllCandidates: async(req, res) => {
-        try {
-            const users = await Candidate.find(req.query);
-            return res.status(HTTPStatus.OK).json({
-                success: true,
-                message: "Candidates fetched successfully",
-                users,
-            });
-        } catch (e) {
-            return res
-                .status(HTTPStatus.BAD_REQUEST)
-                .json({ success: false, message: "Some Error" });
-        }
-    },
+  getAllCandidates: async (req, res) => {
+    try {
+      const { tenant } = req.query;
+      const users = await CandidateModel(tenant).find(req.query);
```

Comment:
@Baalmart I am failing to `get` all candidates. Could this be because the find takes in the tenant key-value pair which results into always an empty array even though there are  candidates for the given org.

Reply:
let me try to reproduce issue now...

Reply:
This has been fixed, please pull the latest commit.

Thanks for the review.